src/Microsoft.CodeAnalysis.Analyzers/UnitTests/InternalImplementationOnlyTests.cs

```diff
@@ -71,35 +102,282 @@ class Foo : IFoo { }
 class Boo : IBar { }";
 
             // Verify no diagnostic since interface is in a friend assembly.
-            VerifyCSharpAcrossTwoAssemblies(source1, source2);
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithoutRoslynSymbols,
+                TestState = { Sources = { source2 } },
+                SolutionTransforms =
+                {
+                    (solution, projectId) =>
+                    {
+                        var dependencyProject = solution.AddProject("DependencyProject", "DependencyProject", LanguageNames.CSharp)
+                            .WithCompilationOptions(solution.GetProject(projectId).CompilationOptions)
+                            .WithParseOptions(solution.GetProject(projectId).ParseOptions)
+                            .WithMetadataReferences(solution.GetProject(projectId).MetadataReferences)
+                            .AddDocument("Test0.cs", source1, filePath: "Test0.cs").Project;
+
+                        return dependencyProject.Solution.AddProjectReference(projectId, new ProjectReference(dependencyProject.Id));
+                    },
+                },
+            }.RunAsync();
         }
 
         [Fact]
-        public void CSharp_VerifyISymbol()
+        public async Task CSharp_VerifyISymbol()
         {
             var source = @"
 // Causes many compile errors, because not all members are implemented.
 class Foo : Microsoft.CodeAnalysis.ISymbol { }
 class Bar : Microsoft.CodeAnalysis.IAssemblySymbol { }
 ";
-            DiagnosticResult[] expected = new[] { GetCSharpExpectedDiagnostic(3, 7, "Foo", "ISymbol"), GetCSharpExpectedDiagnostic(4, 7, "Bar", "ISymbol") };
 
             // Verify that ISymbol is not implementable.
-            VerifyCSharp(source, referenceFlags: ReferenceFlags.None, validationMode: TestValidationMode.AllowCompileErrors, expected: expected);
+            await new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources = { source },
+                    ExpectedDiagnostics =
+                    {
+                        // Test0.cs(3,7): error RS1009: Type Foo cannot implement interface ISymbol because ISymbol is not available for public implementation.
+                        VerifyCS.Diagnostic().WithSpan(3, 7, 3, 10).WithArguments("Foo", "ISymbol"),
+                        // Test0.cs(3,13): error CS0535: 'Foo' does not implement interface member 'ISymbol.Accept(SymbolVisitor)'
+                        DiagnosticResult.CompilerError("CS0535").WithSpan(3, 13, 3, 43).WithArguments("Foo", "Microsoft.CodeAnalysis.ISymbol.Accept(Microsoft.CodeAnalysis.SymbolVisitor)"),
```

Comment:
Is there a test mode where caller can request to skip compiler diagnostic validation? This seems to be a case where forcing compiler diagnostic validation is leading to unnecessary noise in test baseline.

Reply:
There is such a mode, but it should not be used. The solution for this scenario is updating the input code to fully implement the interface with `=> throw null` bodies.